scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1969,7 +2004,7 @@ public void run() {
 	 * @since 3.5
 	 */
 	@NoPublicAPI
-	protected boolean executeInbound(Executor executor, InetSocketAddress peer, LimitedRunnable job) {
+	protected boolean executeInbound(Executor executor, InetSocketAddress peer, LimitedRunnable job, Runnable onError) {
```

Comment:
Maybe, extending `LimitedRunnable`with an `onError(RejectedExecutionException ex)` callback is an alternative to a second `Runnable`. 
Or pass in the `Record` instead of the  second `Runnable`